### PR TITLE
[SPARK-21559][Mesos] remove mesos fine-grained mode

### PR DESCRIPTION
**Pull request by skonto**
## What changes were proposed in this pull request?
Removes mesos fine-grained mode. Specifically:
- Updates docs. 
- Renames spark.mesos.coarse* properties.
- Removes logic for status updates in TaskSchedulerImpl (unique to fine-grained)
- Removes MesosFineGrainedSchedulerBackend, MesosExecutorBackend and related tests.

## How was this patch tested?
Tested with the integration test suite here:
https://github.com/typesafehub/mesos-spark-integration-tests

### CODE REVIEW

**Comment by skonto** [comment]
@ArtRand @susanxhuynh pls review.

**Comment by srowen** [comment]
Overall I trust your judgment if you think it's right to remove this, but I think it would have to wait for Spark 3.0?

**Comment by skonto** [comment]
@srowen ok, yes. I have discussed this with Art and Suzan from Mesosphere and we made the decision to remove it as it is deprecated for so long. I am waiting for their comments here.
In the immediate futurem we will focus on coarse mode and we plan to improve resource utilization & executor allocation per node, also coarse mode + dynamic allocation in general is a good solution.

**Comment by skonto** [comment]
Fixed the test.

File: core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala [hunk 1]
```diff
@@ -784,42 +784,6 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
...
-    assert(taskScheduler.taskIdToExecutorId.isEmpty)
-    assert(taskScheduler.taskIdToTaskSetManager.isEmpty)
-    assert(taskScheduler.runningTasksByExecutors.get("executor0").isEmpty)
-
-    // Check that the executor has been marked as dead
-    assert(!taskScheduler.isExecutorAlive("executor0"))
-    assert(!taskScheduler.hasExecutorsAliveOnHost("host0"))
-    assert(taskScheduler.getExecutorsAliveOnHost("host0").isEmpty)
-  }
-
```
Comment (skonto) [inline core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala]: Test is only needed in the fine-grained case as it is th eonly case where the executor was killed in TaskSchedulerImpl.

**Comment by skonto** [comment]
Jenkins, retest this please

**Comment by skonto** [comment]
@srowen I made a new commit but didn't get a new build...

**Comment by skonto** [comment]
@srowen this test https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/80079/ seems flaky, I run the suite locally and works fine (13 tests finished successfully).

**Comment by skonto** [comment]
Jenkins, retest this please.

**Comment by skonto** [comment]
@ArtRand @susanxhuynh pls review.

**Comment by jiangxb1987** [comment]
@skonto @ArtRand @susanxhuynh Do we still want this?

**Comment by ArtRand** [comment]
@jiangxb1987 yes. I'll work to review this ASAP.

**Comment by imaxxs** [comment]
I work with @ArtRand and @susanxhuynh Fine grained mode has been deprecated for a while. If it is standard procedure to wait till next release and if that is 3.0 we should wait till Spark 3.0 release. @skonto @srowen

**Comment by skonto** [comment]
@rxin I will update the PR.

**Comment by skonto** [comment]
@imaxxs @rxin I think its a good time to remove this, I will update the PR if you are all ok.

**Comment by rxin** [comment]
Go for it.

File: docs/running-on-mesos.md [hunk 2]
```diff
@@ -180,10 +180,7 @@ Note that jars or python files that are passed to spark-submit should be URIs re
 
 # Mesos Run Modes
 
-Spark can run over Mesos in two modes: "coarse-grained" (default) and
-"fine-grained" (deprecated).
-
-## Coarse-Grained
+Spark runs over Mesos in "coarse-grained" mode.
```
Comment (srowen) [inline docs/running-on-mesos.md]: I might add a note here that fine-grained was removed in 3.0.0

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala [hunk 3]
```diff
@@ -81,8 +79,8 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val taskLabels = conf.get("spark.mesos.task.labels", "")
 
   private[this] val shutdownTimeoutMS =
-    conf.getTimeAsMs("spark.mesos.coarse.shutdownTimeout", "10s")
-      .ensuring(_ >= 0, "spark.mesos.coarse.shutdownTimeout must be >= 0")
+    conf.getTimeAsMs("spark.mesos.shutdownTimeout", "10s")
```
Comment (srowen) [inline resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala]: Does this property name change need to be reflected in docs? I wasn't sure.

**Review by srowen — COMMENTED** [review]
@skonto also cool by me. If you resolve the merge conflict, let's do it.

**Comment by srowen** [comment]
@skonto do you want to proceed with this?

**Comment by skonto** [comment]
@srowen sorry missed the msg, yes I will spend some cycles to fix it.

**Comment by vanzin** [comment]
Ping? If you won't update this, please close so someone else can pick it up.

**Comment by felixcheung** [comment]
let's do it!
it's only 2017 though. maybe we should wait until July 2019 its 2nd year anniversary to merge this? (ok, I'm joking)

**Comment by skonto** [comment]
Ok let me see what I can update before closing.

**Comment by skonto** [comment]
closing.

**Comment by felixcheung** [comment]
Don’t we want this?

**Pull request closed without merging**